src/types/uniffi_url.rs

```diff
@@ -0,0 +1,15 @@
+use crate::prelude::*;
+
+#[derive(uniffi::Object)]
```

Comment:
Should probably be:

```rust
#[derive(Debug, PartialEq, Eq, Hash, uniffi::Object)]
#[uniffi::export(Debug, Display, Eq, Hash)]
pub struct FfiUrl {
    url: url::Url,
}

```

Using `UniFFI::export(Eq)` UniFfi will make the resulting Swift Class conform to Swift Equatable, and analogously for Hash etc, and also for Kotlin